cmd/subctl/deploybroker.go

```diff
@@ -80,16 +80,14 @@ func addDeployBrokerFlags() {
 	deployBroker.PersistentFlags().StringVar(&deployflags.ImageVersion, "version", "", "image version")
 
 	deployBroker.PersistentFlags().BoolVar(&deployflags.OperatorDebug, "operator-debug", false, "enable operator debugging (verbose logging)")
-	deployBroker.PersistentFlags().StringVar(&deployflags.BrokerNamespace, "broker-namespace", constants.DefaultBrokerNamespace,
-		"namespace for broker")
 }
 
 func deployBrokerInContext(clusterInfo *cluster.Info, namespace string, status reporter.Interface) error {
-	if err := deploy.Broker(&deployflags, clusterInfo.ClientProducer, status); err != nil {
+	if err := deploy.Broker(namespace, &deployflags, clusterInfo.ClientProducer, status); err != nil {
```

Comment:
Perhaps it would be cleaner to keep the `BrokerNamespace` field?

```suggestion
        deployflags.BrokerNamespace = namespace
	if err := deploy.Broker(&deployflags, clusterInfo.ClientProducer, status); err != nil {
```